elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/scaladsl/ElasticsearchFlow.scala

```diff
@@ -80,6 +80,39 @@ object ElasticsearchFlow {
       .via(stageFlow(indexName, typeName, settings, elasticsearchClient, writer))
       .mapConcat(identity)
 
+  /**
+   * Create a flow to update Elasticsearch with
+   * [[immutable.Seq[akka.stream.alpakka.elasticsearch.WriteMessage WriteMessage]]]s containing type `T`
+   * with `passThrough` of type `C`.
+   * The result status is part of the immutable.Seq[[akka.stream.alpakka.elasticsearch.WriteResult WriteResult]]
+   * and must be checked for successful execution.
+   *
+   * This factory method requires an implicit Spray JSON writer for `T`.
+   */
+  def createBulk[T, C](indexName: String,
+                       typeName: String,
+                       settings: ElasticsearchWriteSettings = ElasticsearchWriteSettings.Default)(
+      implicit elasticsearchClient: RestClient,
+      sprayJsonWriter: JsonWriter[T]
+  ): Flow[immutable.Seq[WriteMessage[T, C]], immutable.Seq[WriteResult[T, C]], NotUsed] =
+    createBulk[T, C](indexName, typeName, settings, new SprayJsonWriter[T]()(sprayJsonWriter))
+
+  /**
+   * Create a flow to update Elasticsearch with
+   * [[immutable.Seq[akka.stream.alpakka.elasticsearch.WriteMessage WriteMessage]]]s containing type `T`
+   * with `passThrough` of type `C`.
+   * The result status is part of the immutable.Seq[[akka.stream.alpakka.elasticsearch.WriteResult WriteResult]]
+   * and must be checked for successful execution.
+   */
+  def createBulk[T, C](indexName: String,
+                       typeName: String,
+                       settings: ElasticsearchWriteSettings,
+                       writer: MessageWriter[T])(
+      implicit elasticsearchClient: RestClient
+  ): Flow[immutable.Seq[WriteMessage[T, C]], immutable.Seq[WriteResult[T, C]], NotUsed] =
+    Flow[immutable.Seq[WriteMessage[T, C]]]
+      .via(stageFlow(indexName, typeName, settings, elasticsearchClient, writer))
```

Comment:
This can be just `stageFlow(...)`.
The onle difference is the implicit client.

Reply:
Good call, Thanks!